### ISSUE TITLE
Running RCU crashes

### ISSUE THREAD

**Issue by gnull** [rvl/pkg-rcu#2]
I've successfully built the RCU with `nix-build --argstr productKey …`, but when I tried to run it I got this error:

```
$ ./result/bin/rcu
running version r2021.001
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile)
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile)
Could not initialize GLX
Fatal Python error: Aborted

Current thread 0x00007f3b57201f80 (most recent call first):
  File "/nix/store/vdv1v3v989f31gvxypcmiqn9m53hhrxn-rcu-r2021.001/lib/python3.8/site-packages/rcu/main.py", line 122 in main
  File "/nix/store/vdv1v3v989f31gvxypcmiqn9m53hhrxn-rcu-r2021.001/bin/.rcu-wrapped", line 9 in <module>
Aborted (core dumped)
```

How do I make it work?

I'm running Nix package manager on Archlinux. If it helps: I'm using Sway (Wayland) system with XWayland.

**Comment by gnull** [comment]
Doing `export QT_XCB_GL_INTEGRATION=none` before running RCU solved the issue for me. Hope this helps.

**Comment by rvl** [comment]
Thanks, it does help. I have only tested on NixOS with xorg-x11.

Just wondering, does `QT_XCB_GL_INTEGRATION=xcb_egl` also work for you?

**Comment by gnull** [comment]
Yes, I understand. It's ok, actually all Xorg apps should work on Wayland through the XWayland compatibility layer. It must be something else that's causing this (maybe something with the way Archlinux configured or something with the library versions).

The other value of the environment variable also crashes:

```
$ QT_XCB_GL_INTEGRATION=xcb_egl rcu
running version r2021.001
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile)
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile)
Could not initialize GLX
Fatal Python error: Aborted

Current thread 0x00007f57c1c69f80 (most recent call first):
  File "/nix/store/vdv1v3v989f31gvxypcmiqn9m53hhrxn-rcu-r2021.001/lib/python3.8/site-packages/rcu/main.py", line 122 in main
  File "/nix/store/vdv1v3v989f31gvxypcmiqn9m53hhrxn-rcu-r2021.001/bin/.rcu-wrapped", line 9 in <module>
Aborted (core dumped)
```

**Comment by rvl** [comment]
OK, thanks for testing.

There is one other thing. The executable `bin/rcu` from this package is actually a shell script wrapper which sets variables for QT and Python. Some of these may potentially conflict with your ArchLinux configuration. I don't know. I just removed hardcoding of `QT_QPA_PLATFORM=xcb` in the wrapper script. It's unnecessary, and perhaps it caused problems for you?

In any case, I added the information which you provided to the README, so thanks.
